### PR TITLE
Backport reverse_view to clean up some code

### PR DESCRIPTION
**Pull request by alexreinking**
I found it very useful to backport `reverse_view` from C++20 for the sake of writing readable backwards loops. Here, I add `reverse_view` to `Util.h`. This allows us to simplify code in a lot of places.

### CODE REVIEW

**Comment by steven-johnson** [comment]
Side note: we should (again) examine if we can move to C++20 for Halide 20. (Google has been at C++20 for a while now. Are there still blockers at Adobe?)

**Review by steven-johnson — APPROVED** [review]
LGTM pending green

File: src/BoundsInference.cpp [hunk 1]
```diff
@@ -328,10 +328,9 @@ class BoundsInference : public IRMutator {
                 }
             }
 
-            const vector<Specialization> &specializations = def.specializations();
-            for (size_t i = specializations.size(); i > 0; i--) {
-                Expr s_cond = specializations[i - 1].condition;
-                const Definition &s_def = specializations[i - 1].definition;
+            for (const auto &s : def.specializations()) {
```
Comment (abadams) [inline src/BoundsInference.cpp:331]: Are you sure it's ok to reverse the order of results here? The ordering leaks out into all the surrounding code.

Comment (alexreinking) [reply src/BoundsInference.cpp]: Oops. Fixed.

File: src/Func.cpp [hunk 2]
```diff
@@ -416,42 +416,42 @@ void check_for_race_conditions_in_split_with_blend(const StageSchedule &sched) {
...
         }
     }
 
     // Now propagate back to all children of the identified root vars, to assert
     // that none of them use a blending tail strategy.
-    for (auto it = sched.splits().begin(); it != sched.splits().end(); it++) {
-        if (it->is_fuse()) {
-            if (parallel.count(it->inner) || parallel.count(it->outer)) {
-                parallel.insert(it->old_var);
+    for (const auto &split : sched.splits()) {
```
Comment (alexreinking) [inline src/Func.cpp:438]: This one is meant to be forward.

**Comment by alexreinking** [comment]
@abadams -- thanks for catching that! I went over all the other loops and they all have `reverse_view`, as they should.

**Comment by abadams** [comment]
Nice cleanup. How complete is it? Did you already grep for all the rbegins/rends, or are there potentially more opportunities?

**Comment by abadams** [comment]
I made it through the new commit and didn't spot any issues.

**Review by abadams — APPROVED** [review]

**Comment by alexreinking** [comment]
Vulkan failures are unrelated

**Pull request merged**